src/compiler/checker.ts

```diff
@@ -29277,6 +29285,28 @@ namespace ts {
             }
         }
 
+        function superCallIsRootLevelInConstructor(superCall: Node, body: Block) {
+            let expressionParent = superCall.parent;
+
+            while (isParenthesizedExpression(expressionParent)) {
+                expressionParent = expressionParent.parent;
+            }
+
+            return expressionParent.parent === body;

```

Comment:
Depending on what you mean by "root level", this check may be inadequate. Consider the following cases, which would return `true` for this function if they were in the constructor body:

```ts
for(super();;) {}
for(a; super();) {} // other arbitrary code can execute in 'a'
for(a; b; super()) {} // other arbitrary code can execute in 'a' and 'b'
for(; ; super()) { break; } // `super()` here will never be executed.
for (const x of super()) {}
while (super()) {}
do {} while (super());
if (super()) {}
switch (super()) {}
```

If the above cases are expected, I'd like to see tests that illustrate the intended behavior.